kotlinx-coroutines-core/common/src/Exceptions.common.kt

```diff
@@ -12,6 +12,9 @@ public expect class CompletionHandlerException(message: String, cause: Throwable
 
 public expect open class CancellationException(message: String?) : IllegalStateException
 
+@Suppress("FunctionName")
+public expect fun CancellationException(message: String?, cause: Throwable?) : CancellationException
```

Comment:
The reasoning to make it public is this: Other people will write cross-platform libs, too, and since we are only allowing `cancel(CancellationException)` they would run into a situation where they have some `cause: Throwable` on their hands and need to cancel some coroutine adding their exception as the cause.